src/services/groups/index.ts

```diff
@@ -75,6 +75,29 @@ export const getGroupsPage = async (
   return await getItem<Paginated<Group>>(url);
 };
 
+export async function getMyGroupsPage(
+  me: User,
+  count: number,
+  startIndex: number = 1,
+  filter?: string
+) {
+  const groupsPromises = [];
+  for(const g of me?.groups ?? []) {
```

Comment:
Why can `me` be undefined? In the function parameters is marked as not optional, so I would except that `me` is always well defined. Also because, how can I fetch my groups if I don't know who `me` is?

Reply:
Replace the for loop with a `map`

```ts
const promises = me.groups.map(group => fetchGroup(group.value);
const groups = await Promise.all(promises)
```